irony.el

```diff
@@ -661,6 +661,17 @@ If no such file exists on the filesystem the special file '-' is
                              (format "%s\n"
                                      (combine-and-quote-strings argv)))))))
 
+(defun irony--send-request-sync (request callback &rest args)
```

Comment:
Thinking about it, the logic isn't quite right here, the check of the callback stack doesn't make so much sense, you should probably register a callback that sets some kind of flag, when the flags is set it means the response has been received.